### ISSUE TITLE
[suggestion] Atomic 32 bits write support

### ISSUE THREAD

**Issue by puzrin** [idubrov/stm32-flash#1]
There are some cases, when it's useful to store 32bit data. For example, `float`. Since memory size is not a problem, this could be default interface, even if smaller size it enougth. But integrity check is required, to verify that write was not interrupted in the middle.

Log record format could be like this:

```
[  1bit commit flag, 15bit tag  ] [ 16bit lo ] [ 16bit hi ]
```

Write sequence is:

1. Write lo/hi
2. Write (tag | 0x1000) <- without commit flag
3. Write (tag & 0x3FFF) <- overwrite the same address with the same data AND commit flag

On scan - check commit flag first to make sure record is valid.

What do you think?

**Comment by idubrov** [comment]
Agree, it could be extended this way. I don't think you really need a commit flag -- you can do the same trick I do now: write data first, then write tag. When looking for empty space, check that all three parts are in the erased state (all bits set).

**Comment by puzrin** [comment]
> write data first, then write tag

This suppose single write to be safe. But power can be lost in the middle of tag write operation, and you can not guarantee that all bits was set to proper value. Having commit flag will give you such guarantee with zero memory cost (tag length is much longer than required in real world).

In theory, it's possible to implement tuneable record size. For example, to store doubles. But i don't think it's useful. IMHO, from application point of view, 32bits is the best size, that cover 99.9% of needs.
